src/pyclts/transcriptionsystem.py

```diff
@@ -268,29 +309,38 @@ def _parse(self, string):
         # we search for aliases and normalize them (as our features system for
         # diacritics may well define aliases
         grapheme, sound = '', ''
-        for dia in [p + EMPTY for p in pre]:
-            feature = self.diacritics[base_sound.type].get(dia, {})
-            if not feature:
-                return UnknownSound(  # noqa: F405
-                    grapheme=nstring, source=string, ts=self)
-            features[self._feature_values[feature]] = feature
-            # we add the unaliased version to the grapheme
-            grapheme += dia[0]
-            # we add the corrected version (if this is needed) to the sound
-            sound += self.features[base_sound.type][feature][0]
+        pre_dia_regex, post_dia_regex = self._diacritic_regexes[base_sound.type]
+
+        for dia in pre_dia_regex.split(pre):
+            # split with match group returns empty strings between matches
+            # empty strings can be ignored
+            if dia:
+                feature = self.diacritics[base_sound.type].get(dia + EMPTY, {})
+                if not feature:
+                    return UnknownSound(  # noqa: F405
+                        grapheme=nstring, source=string, ts=self)
+                features[self._feature_values[feature]] = feature
+                # we add the unaliased version to the grapheme
+                grapheme += dia
+                # we add the corrected version (if this is needed) to the sound
+                sound += self.features[base_sound.type][feature][0]
+
         # add the base sound
         grapheme += base_sound.grapheme
         sound += base_sound.s
-        for dia in [EMPTY + p for p in post]:
-            feature = self.diacritics[base_sound.type].get(dia, {})
-            # we are strict: if we don't know the feature, it's an unknown
-            # sound
-            if not feature:
-                return UnknownSound(  # noqa: F405
-                    grapheme=nstring, source=string, ts=self)
-            features[self._feature_values[feature]] = feature
-            grapheme += dia[1]
-            sound += self.features[base_sound.type][feature][1]
+        for dia in post_dia_regex.split(post):
```

Comment:
I considered refactoring these two nearly identical blocks to avoid repetition, but since I am not an usual contributor to pyclts, I opted for staying as close as possible to the current implementation.